### PR TITLE
Implementing scenario duplication endpoint

### PR DESCRIPTION
**Pull request by elliotmassen**
This PR adds a duplication endpoint for scenarios, and is dependent on https://github.com/opendialogai/core/pull/524 & https://github.com/opendialogai/opendialog-design-system/pull/174. The endpoint will make a copy of the original scenario and append " Copy" to its name (and add a number if that already exists).﻿

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
         "laravel/tinker": "^2.0",
         "laravel/ui": "^3.0",
         "maennchen/zipstream-php": "^2.0",
-        "opendialogai/core": "1.x-dev",
+        "opendialogai/core": "dev-feature/OPNDLG-546--duplicate-scenario",
```
Comment (elliotmassen) [inline composer.json:19]: This will need to be updated before merging.

File: app/Http/Controllers/API/ScenariosController.php [hunk 2]
```diff
@@ -166,6 +164,7 @@ private function createDefaultConversations(Scenario $scenario): Scenario
         $scenario->addConversation($noMatchConversation);
 
         $scenario = ConversationDataClient::addFullScenarioGraph($scenario);
```
Comment (stu-greenshoots) [inline app/Http/Controllers/API/ScenariosController.php:166]: We overwrite the scenario here - do we need to fetch it again?

File: app/Http/Requests/ConversationObjectDuplicationRequest.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
+<?php
```
Comment (stu-greenshoots) [inline app/Http/Requests/ConversationObjectDuplicationRequest.php:1]: Like that this is reusable like this 👍

**Review by stu-greenshoots — COMMENTED** [review]
Works for me - just a little question about the flow in the controller as its not clear to me

File: app/Http/Controllers/API/ScenariosController.php (hunk 2 above)
Comment (elliotmassen) [reply app/Http/Controllers/API/ScenariosController.php]: Good catch - that's fixed now.

**Review by stu-greenshoots — APPROVED** [review]

**Pull request merged**